### DISCUSSION TITLE
Advice for batching SELECT statements?

### DISCUSSION THREAD

**Discussion by emchristiansen** [SeaQL/sea-orm#1514] [Q&A] [answered]
Does SeaQL / SeaORM support batched SELECT statements?
If not, does anyone have tips on executing many nearly-identical queries?

I need to execute ~10K SELECT statements / second.
The statements are parameterized by a single value and are thus nearly identical and should re-use the same execution plans.

I'm currently using `UNION ALL` and constructing a giant "unrolled" query, but that turns out to be absurdly slow.
I'm in the process of looking into it, but I expect nearly all the overhead is in the SQL compiler and planner.

Instead, I'd like to tell the DB "Here's a parameterized SELECT statement and here are 10K parameter values. Execute them all and return.".

Is this something I can do in SeaQL / SeaORM (I don't see anything like `select_many` in the API)?

FYI, [I asked for advice on the same problem here](https://stackoverflow.com/questions/75587564/how-to-execute-the-same-select-statement-for-every-element-in-a-set-and-union-t).
I was hoping I could express the batching directly in SQL, but it appears not to be cleanly supported.
(Perhaps I should be using pre-compiled DB scripts?)

**Comment by tyt2y3** [comment, marked as answer]
I think you are better off create a view and select against it.
Although I don't think the protocol can handle up to 10k parameters.
And I don't think melding everything together would pan out well.
You are basically limiting yourself by the serialization throughput of one connection,
which will have giant latency towards the end.
Without more context, my best advice is to rethink the architecture and use Redis as a caching layer, etc.

**Reply by emchristiansen** [reply]
Thanks for the tips.
I ended up rewriting the schema and app logic as a workaround.

FYI, I am able to get ~100K QPS for simple key lookups using TiDB on a 8 core machine.
So instead of using Redis, I amended my schema and I'm now basically using TiDB both for "real SQL" and as a key-value cache.
